neuralprophet/time_net.py

```diff
@@ -987,6 +988,73 @@ def denormalize(self, ts):
     def train_dataloader(self):
         return self.train_loader
 
+    def get_regressor_coefficients(self):
+        """
+        Retrieves the coefficients for future regressors, lagged regressors, and events.
+
+        Note: The average weight calculation is performed to get a single representative
+        value of the coefficient for a given regressor when there are multiple forecasts
+        or hidden layers.
+
+        Returns
+        -------
+            pd.DataFrame: A DataFrame containing the following columns:
+                - regressor: Name of the regressor.
+                - regressor_mode: Mode of the regressor ('additive' or 'multiplicative').
+                - coef: Average coefficient value for the regressor.
+        """
+        coefficients = []
+
+        # Helper function to calculate the average weight
+        def calculate_average_weight(weight):
+            if weight.ndim == 2:
+                return weight.mean(axis=0).mean(axis=0)
+            elif weight.ndim == 1:
+                return weight.mean(axis=0)
+            else:
+                return weight
+
+        # Future Regressors
+        if self.config_regressors is not None and self.config_regressors.regressors is not None:
+            for name, config in self.config_regressors.regressors.items():
+                regressor_component = self.future_regressors
+                if config.mode == "additive" and "additive" in regressor_component.regressor_params:
+                    coef = regressor_component.regressor_params["additive"].data.cpu().numpy()
+                elif config.mode == "multiplicative" and "multiplicative" in regressor_component.regressor_params:
+                    coef = regressor_component.regressor_params["multiplicative"].data.cpu().numpy()
+                else:
+                    coef = [None]
```

Comment:
Are there cases where this should end up being `None`?

Reply:
No, that should not be the case.